services/121-service/src/cronjob/cronjob.service.ts

```diff
@@ -1,184 +1,135 @@
 import { HttpService } from '@nestjs/axios';
-import { Injectable } from '@nestjs/common';
+import { HttpStatus, Injectable } from '@nestjs/common';
 import { Cron, CronExpression } from '@nestjs/schedule';
 import { AxiosResponse } from '@nestjs/terminus/dist/health-indicator/http/axios.interfaces';
 
-import { CustomHttpService } from '@121-service/src/shared/services/custom-http.service';
+import {
+  CustomHttpService,
+  Header,
+} from '@121-service/src/shared/services/custom-http.service';
 import { AxiosCallsService } from '@121-service/src/utils/axios/axios-calls.service';
 
 function shouldBeEnabled(envVariable: string | undefined): boolean {
   return !!envVariable && envVariable.toLowerCase() === 'true';
 }
 
+type cronReturn = Promise<
+  | {
+      url: string;
+      responseStatus: HttpStatus;
+    }
+  | false
+>;
+
 @Injectable()
 export class CronjobService {
   private httpService = new CustomHttpService(new HttpService());
   private axiosCallsService = new AxiosCallsService();
+  private headers: Header[];
+  private baseUrl: string;
```

Comment:
I am not sure if headers and baseUrl should be a class attribute if they are set again every time a cronjob is ran. So for example if there was like a `getHeaders` function that would get the headers from the class attribute and re-authenticates if they are expired it would make sense to me to store it as a class attribute as than the 'headers' are shared between cronjob function. However if they are re-initiated every-time it think it's clearer if they are the return of a function instead of a class attribute.

I think the baseurl could be class attribute but than I think it makes more sense to set it 'onModuleInit'. But I think it's also fine to just call the getBaseUrl method everytime.

What do you think?

Reply:
Agreed, I'll push some changes to remove these class properties. I've had to change some tests too to be able to do this.